packages/core/src/routes/hook.ts

```diff
@@ -21,23 +24,59 @@ export default function hookRoutes<T extends AuthedRouter>(
     }
   );
 
+  router.get(
+    '/hooks/:id',
+    koaGuard({
+      params: z.object({ id: z.string() }),
+      response: Hooks.guard,
+      status: [200, 404],
+    }),
+    async (ctx, next) => {
+      const {
+        params: { id },
+      } = ctx.guard;
+
+      ctx.body = await findHookById(id);
+
+      return next();
+    }
+  );
+
   router.post(
     '/hooks',
-    koaGuard({ body: Hooks.createGuard.omit({ id: true }), response: Hooks.guard, status: 200 }),
+    koaGuard({
+      body: Hooks.createGuard
+        .omit({ id: true, name: true, signingKey: true, enabled: true, events: true })
+        .extend({
+          // Note: ensure the user will not create a hook with an empty name.
+          name: string().min(1).optional(),
+          events: hookEventsGuard.nonempty().optional(),
+        }),
+      response: Hooks.guard,
+      status: [201, 422],
+    }),
     async (ctx, next) => {
+      const { event, events, ...rest } = ctx.guard.body;
+      assertThat(events ?? event, new RequestError({ code: 'hook.missing_events', status: 422 }));
+
       ctx.body = await insertHook({
+        ...rest,
         id: generateStandardId(),
-        ...ctx.guard.body,
+        signingKey: generateStandardId(),
+        enabled: true,
+        ...conditional(event && { event }),
+        ...conditional(events && { events: deduplicate(events) }),
       });
+      ctx.status = 201;
 
       return next();
     }
   );
 
-  router.get(
-    '/hooks/:id',
+  router.post(
+    '/hooks/:id/signing-key',
     koaGuard({
-      params: z.object({ id: z.string().min(1) }),
+      params: z.object({ id: z.string() }),
       response: Hooks.guard,
       status: [200, 404],
```

Comment:
201 as well?

Reply:
According to the [definition](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/201) of the `201` status code, maybe 200 is more suitable, since no hook is newly created by this API.

Reply:
signingKey is a resource of hook itself?

Reply:
Change the API method to patch since it's an update action, and return the whole hook data.